articles/plagiarism-detection/index.md

```diff
@@ -0,0 +1,402 @@
+
+##### 1. Installing Dependencies
+You we need to install the following: `Pandas` and `scikit-learn`
+
+```Bash
+$ conda install scikit-learn pandas
+```
+
+##### 2. Verification
+To ensure that the packages are installed properly open up your python interpreter and run the follow code.
+
+``` python
+# loading python modules
+import sklearn
+import pandas as pd
+
+# Verifying version
+sklearn.show_versions()
+pd.show_versions()
+```
+> You may need to install pytest and hypothesis for pandas to run command `pd.show_versions()`
+> Just run command`$ conda install pytest hypothesis`
+
+#### Code Walkthrough
+Now that we have everything set up it's time to code!
+
+##### 1. Load Python Modules
+Let's first import important modules
+
+```python
+# Load Python Modules
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+
+import pandas as pd
+
+```
+
+##### 2. Defining Data Set
+Here we define the data. I have set up a list of `tuples(name, data)`.
+
+```python
+# define data
+corpus = [
+        ('doc_1', 'This is the first document.'),
+        ('doc_2', 'This document is the second document.'),
+        ('doc_3' ,'And this is the third one.'),
+        ('doc_4', 'Is this the first document?')
+    ]
+
+```
+
+##### 3. Process the Data/Helper Functions
+*Using Count Vectorize*
+Now we need to process the data. Here is a neat trick to separate the names and data into lists.
+
+```python
+# split doc_names and doc_data
+doc_names, doc_data = zip(*corpus)
+```
+Output:
+```python
+doc_names -> ('doc_1', 'doc_2', 'doc_3', 'doc_4')
+```
+```python
+doc_data -> ('This is the first document.', \
+            'This document is the second document.', \
+            'And this is the third one.', \
+            'Is this the first document?')
+```
+
+##### 4. Vectorize Data
+This is the most crucial step. We need to convert the data into a vector space. Luckily `sklearn` as a function called `CountVectorizer()` that will do the heavy lifting.
+
+```python
+#create an instance of the class Countvectorizer that converts a collection of text document to a matrix of token counts
+vectorizer = CountVectorizer()
+
+
+# vectorize doc_data
+document_term_matrix = vectorizer.fit_transform(doc_data).toarray() -> array([[0, 1, 1, 1, 0, 0, 1, 0, 1],
+                                                                              [0, 2, 0, 1, 0, 1, 1, 0, 1],
+                                                                              [1, 0, 0, 1, 1, 0, 1, 1, 1],
+                                                                              [0, 1, 1, 1, 0, 0, 1, 0, 1]])
+
+# returns full list of tokenized words
+tokenized_words =  vectorizer.get_feature_names() -> ['and', 'document', 'first', 'is', 'one', 'second', 'the', 'third', 'this']
+
+# output pandas table document_term_matrix
+df_document_term_matrix = pd.DataFrame(data=document_term_matrix,
```

Comment:
Python is a whitespace matter language. When we compile this into a website, it will automatically add tabs to match how you typeset it. This breaks copy paste. Please put this on one line.